candidates/models/versions.py

```diff
@@ -172,3 +176,71 @@ def revert_person_from_version_data(person, person_extra, version_data):
         update_twitter_user_id(person)
     except TwitterAPITokenMissing:
         pass
+
+def version_timestamp_key(version):
+    return datetime.strptime(version['timestamp'], '%Y-%m-%dT%H:%M:%S.%f')
+
+def is_a_merge(version):
+    m = re.search(
+        r'^After merging person (\d+)', version['information_source'])
+    if m:
+        return m.group(1)
+    return None
+
+def get_versions_parent_map(versions_data):
+    version_id_to_parent_ids = {}
+    if not versions_data:
+        return version_id_to_parent_ids
+    canonical_person_id = versions_data[-1]['data']['id']
+    ordered_versions = sorted(versions_data, key=version_timestamp_key)
+    person_id_to_ordered_version = defaultdict(list)
+    # Divide all the version with the same ID into separate ordered
+    # lists, and record the parent of each version that we get from
+    # doing that:
+    for version in ordered_versions:
+        version_id = version['version_id']
+        person_id = version['data']['id']
+        versions_for_person_id = person_id_to_ordered_version[person_id]
```

Comment:
I see what you mean about the `_id` suffix, but my hope is always that be keeping the distinction between `person` and `person_id` in variable names, a certain type of error will jump out at you that wouldn't otherwise. I'm not sure if that actually works or not :/

I'll change the variable name to `person_to_ordered_versions`.